synapse/event_auth.py

```diff
@@ -50,6 +50,18 @@ def check(room_version, event, auth_events, do_sig_check=True, do_size_check=Tru
     if not hasattr(event, "room_id"):
         raise AuthError(500, "Event has no room_id: %s" % event)
 
+    room_id = event.room_id
+
+    # I'm not really expecting to get auth events in the wrong room, but let's
+    # sanity-check it
+    for auth_event in auth_events.values():
+        if auth_event.room_id != room_id:
+            raise Exception(
```

Comment:
Do we want to 500 or 403?

Reply:
given that I think the problem has happened somewhere else if we get this far, I think a 500 is more appropriate. Consider it as an assertion.

Tbh I think that our habit of trying to decide what HTTP error codes we should return in the depths of utility functions is an anti-pattern.

Reply:
Fair